src/bokeh/plotting/geo_feature.py

```diff
@@ -0,0 +1,151 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# External imports
+import numpy as np
+
+try:
+    import cartopy.crs as ccrs
+    import cartopy.feature as cfeature
+except ModuleNotFoundError as e:
+    raise ModuleNotFoundError(
+        "The module 'geo_feature' requires the optional dependency 'cartopy'."
+        "To use this module, please install 'cartopy'.",
+    ) from e
+
+
+def _collect_line_geometries(projection, geometries_collection):
+    xs = []
+    ys = []
+    for geometry in geometries_collection.geometries():
+        lines = projection.project_geometry(geometry, src_crs=ccrs.PlateCarree())
+        for line in lines.geoms:
+            x, y = np.array(line.xy)
+            xs.append(x)
+            ys.append(y)
+    return xs, ys
+
+
+def _collect_polygon_geometries(projection, geometries_collection, **kwargs):
+    xs = []
+    ys = []
+    color = kwargs.get("color")
+    color_selection = isinstance(color, (list, tuple))
+    selected_colors = []
+    if color_selection:
+        n_color = len(color)
+
+    for i, geometry in enumerate(geometries_collection.geometries()):
+        projected_geometrie = projection.project_geometry(geometry, src_crs=ccrs.PlateCarree())
+        for geom in projected_geometrie.geoms:
+            x, y = np.array(geom.exterior.xy)
+            x_geom = [x]
+            y_geom = [y]
+            for interior in geom.interiors:
+                x, y = np.array(interior.coords.xy)
+                x_geom.append(x)
+                y_geom.append(y)
+
+            xs.append([x_geom])
+            ys.append([y_geom])
+            if color_selection:
+                selected_colors.append(color[i%n_color])
+    if color_selection:
+        kwargs.update(color=selected_colors)
+    return xs, ys, kwargs
+
+
+def _collect_lines_from_ploygons(polygon_xs, polygon_ys):
+    line_xs = []
+    line_ys = []
+    for xs, ys in zip(polygon_xs, polygon_ys):
+        line_xs.append(xs[0][0])
+        line_ys.append(ys[0][0])
+    return line_xs, line_ys
```

Comment:
This should be located in a private section (header is missing). Also public section is missing.